include/deal.II/lac/diagonal_matrix.h

```diff
@@ -27,6 +27,19 @@ DEAL_II_NAMESPACE_OPEN
  * VectorType::scale, so the template vector class needs to provide a
  * @p scale() method.
  *
+ * When using this class with ConstraintsMatrix::distribute_local_to_global(),
+ * the underlying vector needs to be a ghost vector as during assembly there
+ * will be locally owned cell where some of the DoFs are not owned by the
```

Comment:
Please reword to something like: "..., the underlying vector needs to provide write access to all entries referenced by the underlying cell in an assembly process. This means that this class also needs access to ghost entries that are owned by other processors than the calling one."